tmail-backend/jmap/extensions/src/main/java/com/linagora/tmail/james/jmap/contact/OpenPaasContactsConsumer.java

```diff
@@ -92,29 +96,43 @@ public Flux<AcknowledgableDelivery> delivery() {
 
     private void messageConsume(AcknowledgableDelivery ackDelivery, String messagePayload) {
         Mono.just(messagePayload)
-            .map(message -> gson.fromJson(message, OpenPaasContactMessage.class))
+            .<ContactAddedRabbitMqMessage>handle((message, sink) -> {
+                try {
+                    sink.next(objectMapper.readValue(message, ContactAddedRabbitMqMessage.class));
+                } catch (JsonProcessingException e) {
+                    sink.error(new RuntimeException(e));
+                }
+            })
             .handle((msg, sink) -> {
                 handleMessage(msg, sink);
                 ackDelivery.ack();
-            });
+            })
+            .onErrorResume(e -> {
+                LOGGER.error("Failed to consume OpenPaaS added contact message", e);
+                return Mono.empty();
+            }).subscribe();
     }
 
-    private void handleMessage(OpenPaasContactMessage message, SynchronousSink<Object> sink) {
+    private void handleMessage(ContactAddedRabbitMqMessage contactAddedMessage, SynchronousSink<Object> sink) {
+        LOGGER.info("Consumed jCard object message: {}", contactAddedMessage);
         try {
-            String firstname = message.getFirstname();
-            Username username = Username.of(message.getUsername());
-            String lastname = message.getLastname();
-            MailAddress mailAddress = new MailAddress(message.getEmail());
-            Mono.from(contactSearchEngine.index(AccountId.fromUsername(username),
-                new ContactFields(mailAddress, firstname, lastname))).block();
+            jCardObject jCardObject = contactAddedMessage.vcard();
+            String fullname = jCardObject.fn();
+            MailAddress mailAddress = new MailAddress(jCardObject.email());
+            Username emailAsUsername = Username.fromMailAddress(mailAddress);
+
+            Mono.from(contactSearchEngine.index(AccountId.fromUsername(emailAsUsername),
+                new ContactFields(mailAddress, fullname, ""))).block();
```

Comment:
Do not block here but return the mono and compose it (`.flatMap` ?) in caller code.